spring-cloud-task-samples/partitioned-batch-job/src/main/java/io/spring/JobConfiguration.java

```diff
@@ -69,6 +71,9 @@
 	@Autowired
 	private ConfigurableApplicationContext context;
 
+	@Autowired
```

Comment:
I feel like this should be a param for the `partitionHandler` bean config instead.

Or  the `@Autowired` could contain `required=false` so that if the conditional config does not find the `MavenResourceLoader`, no other `DelegatingResourceLoader` bean needs to be defined. Instead, a `DefaultResourceLoader` could be used for the `partitionHandler` if the autowired field is null.